backend/controllers/whereabouts/scheduledMoves.ts

```diff
@@ -24,6 +21,98 @@ export default ({ prisonApi }) => {
           text: values.description,
         }))
         .sort(sortTextAlphabetically),
+      courtEvents,
+    })
+
+  const index = async (req, res) => {
+    const { userDetails } = req.session
+    const { activeCaseLoadId } = userDetails
+    const date = req.query?.date || moment().format('DD/MM/YYYY')
+
+    const movementReasons = await prisonApi.getMovementReasons(res.locals)
+    const agencyDetails = await prisonApi.getAgencyDetails(res.locals, activeCaseLoadId)
+
+    const transfers = await prisonApi.getTransfers(res.locals, {
+      fromDateTime: isoDateTimeStartOfDay(date, 'DD/MM/YYYY'),
+      toDateTime: isoDateTimeEndOfDay(date, 'DD/MM/YYYY'),
+      agencyId: activeCaseLoadId,
+      courtEvents: true,
+      transferEvents: true,
+      releaseEvents: true,
+    })
+
+    const offenderNumbers = [
+      ...transfers.courtEvents.map((ce) => ce.offenderNo),
+      ...transfers.transferEvents.map((te) => te.offenderNo),
+      ...transfers.releaseEvents.map((re) => re.offenderNo),
+    ]
+
+    if (offenderNumbers.length === 0) {
+      return renderTemplate(res, {
+        date,
+        agencyDetails,
+        movementReasons,
+        courtEvents: [],
+      })
+    }
+
+    const searchResult = await offenderSearchApi.getPrisonersDetails(res.locals, offenderNumbers)
+
+    const prisonerProperty =
+      (await Promise.all(
+        searchResult.flatMap((sr) =>
+          prisonApi.getPrisonerProperty(res.locals, sr.bookingId).then((property) =>
+            property.map((prop) => ({
+              containerType: prop.containerType,
+              userDescription: formatPropertyDescription(prop.location.userDescription),
+              bookingId: sr.bookingId,
+              prisonerNumber: sr.prisonerNumber,
+            }))
+          )
+        )
+      )) || []
+
+    const prisonerCommonShared = searchResult.map((details) => {
+      const alertCodes = details.alerts
+        .map((alert) => alert.alertCode)
+        .filter((alert) => relevantAlertsForTransfer.includes(alert))
+
+      const relevantAlertLabels = alertFlagLabels.filter((flag) =>
```

Comment:
Won't `relevantAlertsForTransfer` just include codes in `alertFlagLabels`? So you just need to have another map above rather than 2 statements?!

Reply:
There are three data sets. Prisoner alerts, applicable alerts for transfers and the view model data that contains things like image location and correct css class. I'll try improve the readability in the next PR as this is a work in progress.